broadinstitute/single_cell_portal_core
# Adding high-level metadata summary to search results (SCP-5845)

#### BACKGROUND
A recent [Zendesk request](https://broadinstitute.zendesk.com/agent/tickets/322279) was submitted by a user looking for ways to export or preview high-level information about studies in search results.  Currently, the only way to filter studies is through the search interface, but only on the filter values selected.  There is no way to glean additional high-level information (e.g. species, disease, etc.) about search results without loading each study individually and inspected the contents of said annotations.  The requesting user referenced the [Genomic Data Commons cohort builder](https://portal.gdc.cancer.gov/analysis_page?app=CohortBuilder&tab=general) as an example.

#### CHANGES
This update now adds a table to all search results (both SCP- and HCA-based) that contains the entries from 5 metadata convention based annotations: `disease__ontology_label`, `organ__ontology_label`, `species__ontology_label`, `library_preparation_protocol__ontology_label`, and `sex`.  This data is now present in the search API response for any request.  Each study will show up to 2 values per annotation, with the rest hidden behind a `{n} more...` label with a tooltip containing the rest of the labels.  Any studies lacking the information for a given column will display `unspecified`.  These result tables are shown both in study- and gene-based searches.

Examples of results:
![Screenshot 2024-10-21 at 3 25 12 PM](https://github.com/user-attachments/assets/c8ab7ab9-5588-4898-ad8c-24233d9b1183)

![Screenshot 2024-10-21 at 3 24 43 PM](https://github.com/user-attachments/assets/6a178df6-c20f-48b6-90d8-43d8fc63754b)

`more...` tooltip
![Screenshot 2024-10-21 at 3 27 02 PM](https://github.com/user-attachments/assets/b9b8dd28-4c6a-4b05-9fda-1e98df8cca30)

Example in global gene search:
![Screenshot 2024-10-21 at 3 25 59 PM](https://github.com/user-attachments/assets/7e4adfdd-06f5-4fb2-9083-ea52ed7af8d0)


This can also be viewed as potential preparatory work for a "cohort-builder" type interface, though this would only be for study-level data.  Cell-level cohorts would likely have more power and utility, but SCP lacks a true cell-level search interface, though it would be possible for metadata convention compliant studies.

#### MANUAL TESTING
1. Boot as normal and confirm default results contain metadata information
2. Run any search (facet- or text-based) and confirm that normal functionality is not affected
3. Run a global gene search and confirm the metadata table renders above the violin plot as well
